### ISSUE TITLE
Fixed or minimal version for python dependencies 

### ISSUE THREAD

**Issue by mdiazmel** [aramis-lab/clinica#60]
Sometimes when installing another python package that has common requirements with Clinica (let's say X), package managers like `pip` or `conda` could not solve dependencies easily. This is because Clinica's requirements correspond to fixed versions (difficulty level depends if Clinica was installed first or after the X package). 
It's acceptable to have fixed version for some packages (those ones used mainly  by Clinica) but but for some others it should be nice to have  minimal versions (i.e. scikit-learn, numpy, jinja2, pydot, six).

**Comment by alexandreroutier** [comment]
Hi,

I think the freezing of the `requirements.txt` file was due to the fact that implicit dependancies (e.g. `networkx` if I remember well) were causing issues due to new release (e.g. old functions were no longer available). 

With the CI, we should be able to 'unfreeze' several Clinica dependancies. If a package follows semantic versioning like Jinja2, we could replace `jinja2 == 2.10.1` by  `jinja2 >= 2.10.1` & `jinja2 < 3.0.0`.

I started to list packages month ago: https://docs.google.com/spreadsheets/d/1Zf7y1T4-AI7i2XAgD1jqoXC6BxUQsUO1f8D5gwQRVZM/edit#gid=0

If we add information on what are the packages following semantic versioning, we should be able to update `requirements.txt`.

Alex
